File: src/WeightTracker.py
import os

def CreateFileIfNotExists(strFile):
	if(os.path.isfile(strFile))
		return True
	try:
		f = open(strFile,"w")
		f.write("")
		f.close()
	except Exception as e:
		return False
	return True

class WeightData:
	def __init__(self,strFileName):
		self.strFileName = strFileName
		self.weights = []

	def ReadWeights(self):
		if(!CreateFileIfNotExists(self.strFileName)):
			print("Failed to create file")
			return False
		f = None
		try:
			f = open(self.strFileName,"r")
		except Exception as e:
			print("Failed to open file for reading")
			return False

		for l in f:
			try:
				n = int(l)
				self.weights.append(n)
			except Exception as e:
				pass
		f.close()
		return True

	def IsValidWeight(self, nSt, nLbs):
		return nSt > 0 and nLbs >= nLbs and nLbs < 14

	def RemoveWeight(self,nIndex):
		if nIndex < 1 or nIndex > len(self.weights):
			print("Enter a valid index!")
			return False
		del self.weights[nIndex-1]
		f = None
		try:
			f = open(self.strFileName,"w")
			for i in weights:
				f.write(i)
			f.close()
		except Exception as e:
			print("There was a problem writing the data file")
			return False
		return True

File: src/WeightTracker.cpp
#include <iostream>
#include "WeightData.h"
using namespace std;

int Usage(){
  cout
  << "Usage:" << endl
  << "\tWeightTracker add 10 12\tRecord a new record (10st 12lbs)" << endl
  << "\tWeightTracker rem 10   \tRemoves 10th record" << endl
  << "\tWeightTracker rec 10   \tShows 10th record" << endl
  << "\tWeightTracker rec      \tShows current stats" << endl;
  return 0;
}

int main(int argc, char ** argv){
  if(argc < 2)
    return Usage();

  // Create an instance of WeightData to interface with the file which holds the weights
  WeightData weigh("Weight.data");

  if(!weigh.ReadWeights())
    return 1;


  switch(argc){
    case 2:
      if(string(argv[1]) == string("rec")){
        cout << weigh.GetRecordFormatted() << endl;
				break;
			}
		case 3:
			if(string(argv[1]) == string("rec")){
				try{
					int nIndex = stoi((string)argv[2]);
					cout << weigh.GetRecordFormatted(nIndex) << endl;
					break;
				}catch(exception e) {
					Usage();
					return 1;
				}
			}
			if(string(argv[1]) == string("rem")){
				try{
					int nIndex = stoi((string)argv[2]);
					if(weigh.RemoveRecord(nIndex))
						break;
					else
						return 1;
				} catch(exception e){
					Usage();
					return 1;
				}
			}
		case 4:
			if(string(argv[1]) == string("add")){
				int nSt,nLbs;
				try{
					nSt = stoi((string)argv[2]);
					nLbs = stoi((string)argv[3]);
					if(!WeightData::IsValidWeight(nSt,nLbs)){
						cout << "Invalid Weight!" << endl;
						return 1;
					}
					weigh.AddWeight(nSt,nLbs);
					break;
				} catch(exception e){
					Usage();
					return 1;
				}
			}
    default:
      return Usage();
  }
  return 0;
}
